examples/conway/src/main.rs

```diff
@@ -66,13 +66,29 @@ fn main() -> Result<(), Error> {
                     let (dx, dy) = input.mouse_diff();
                     let prev_x = mx - dx;
                     let prev_y = my - dy;
-                    let dpx = hidpi_factor as f32;
-                    let (w, h) = (p_width as f32 / dpx, p_height as f32 / dpx);
-                    let mx_i = ((mx / w) * (SCREEN_WIDTH as f32)).round() as isize;
-                    let my_i = ((my / h) * (SCREEN_HEIGHT as f32)).round() as isize;
-                    let px_i = ((prev_x / w) * (SCREEN_WIDTH as f32)).round() as isize;
-                    let py_i = ((prev_y / h) * (SCREEN_HEIGHT as f32)).round() as isize;
-                    ((mx_i, my_i), (px_i, py_i))
+
+                    let (mx_i, my_i) = pixels
+                        .window_pos_to_pixel((mx as f64, my as f64))
+                        .unwrap_or_else(|(x, y)| {
+                            (
+                                x.max(0).min(SCREEN_WIDTH as isize - 1) as usize,
+                                y.max(0).min(SCREEN_HEIGHT as isize - 1) as usize,
+                            )
+                        });
+
+                    let (px_i, py_i) = pixels
+                        .window_pos_to_pixel((prev_x as f64, prev_y as f64))
+                        .unwrap_or_else(|(x, y)| {
+                            (
+                                x.max(0).min(SCREEN_WIDTH as isize - 1) as usize,
+                                y.max(0).min(SCREEN_HEIGHT as isize - 1) as usize,
+                            )
+                        });
```

Comment:
I definitely need outside->inside_edge clipping. I use mouse movements for drawing pixels to a grid basically, and it would be too hard to place on the edges if you had to be exactly in the window.

Reply:
I believe that it would be useful to have an example that handles it that way, as that would be one of the two common ways that people would want to handle it (clamping to screen / ignoring locations outside).  Ignoring locations outside is easier to implement, so I believe it would be useful to have an example of clamping to the edge of the screen.

Reply:
Ok, going over this again. This example can be greatly simplified by providing a clamp method on the `Pixels` struct. And (at least this example) can be simplified even further when the pixel position returned is `Ok((isize, isize))`:

```rust
let (mx_i, my_i) = pixels
    .window_pos_to_pixel((mx as f64, my as f64))
    .unwrap_or_else(|pos| pixels.clamp_pixel_pos(pos));

let (px_i, py_i) = pixels
    .window_pos_to_pixel((prev_x as f64, prev_y as f64))
    .unwrap_or_else(|pos| pixels.clamp_pixel_pos(pos));
```

I'm inclined to just do this as a good tradeoff between simplicity and necessity.